mmv1/products/vertexai/Models.yaml

```diff
@@ -0,0 +1,419 @@
+            description: |-
+              The distance measure used in nearest neighbor search. The value must be one of the followings:
+              * SQUARED_L2_DISTANCE: Euclidean (L_2) Distance
+              * L1_DISTANCE: Manhattan (L_1) Distance
+              * COSINE_DISTANCE: Cosine Distance. Defined as 1 - cosine similarity.
+              * DOT_PRODUCT_DISTANCE: Dot Product Distance. Defined as a negative of the dot product
+            default_value: 'DOT_PRODUCT_DISTANCE'
+          - type: String
+            name: 'featureNormType'
+            description: |-
+              Type of normalization to be carried out on each vector. The value must be one of the followings:
+              * UNIT_L2_NORM: Unit L2 normalization type
+              * NONE: No normalization type is specified.
+            default_value: 'NONE'
+            ignore_read: true
+          - type: NestedObject
+            name: 'algorithmConfig'
+            description:
+              The configuration with regard to the algorithms used for efficient
+              search.
+            properties:
+              - type: NestedObject
+                name: 'treeAhConfig'
+                exactly_one_of:
+                  - treeAhConfig
+                  - bruteForceConfig
+                description: |-
+                  Configuration options for using the tree-AH algorithm (Shallow tree + Asymmetric Hashing).
+                  Please refer to this paper for more details: https://arxiv.org/abs/1908.10396
+                properties:
+                  - type: Integer
+                    name: 'leafNodeEmbeddingCount'
+                    description:
+                      Number of embeddings on each leaf node. The default value
+                      is 1000 if not set.
+                    default_value: 1000
+                  - type: Integer
+                    name: 'leafNodesToSearchPercent'
+                    description: |-
+                      The default percentage of leaf nodes that any query may be searched. Must be in
+                      range 1-100, inclusive. The default value is 10 (means 10%) if not set.
+                    default_value: 10
+              - type: NestedObject
+                name: 'bruteForceConfig'
+                allow_empty_object: true
+                send_empty_value: true
+                properties: []
+                exactly_one_of:
+                  - treeAhConfig
+                  - bruteForceConfig
+                description: |-
+                  Configuration options for using brute force search, which simply implements the
+                  standard linear search in the database for each query.
+  - type: String
+    name: 'sourceModel'
+    default_from_api: true
+    at_least_one_of:
+      - sourceModel
+    description: |
+      The resource name of the Model to copy. That Model must be in the same Project
+  - type: String
+    name: 'modelId'
+    default_from_api: true
```

Comment:
Do we need to still include `default_from_api`? having ignore_read should be enough. From the docs:

`ignore_read`: If true, the provider sets the field’s value in the resource state based only on the user’s configuration. If false or unset, the provider sets the field’s value in the resource state based on the API response

I also included `immutable: true` since users are unable to edit the model_id

post_create has been included.

Reply:
Addressed in [b32aafa](https://github.com/GoogleCloudPlatform/magic-modules/pull/12074/commits/b32aafa4db1229fab6f4f5fdae66d6f910a3ce2b)